app/src/main/java/lt/vilnius/tvarkau/views/adapters/ProblemImagesPagerAdapter.kt

```diff
@@ -0,0 +1,109 @@
+package lt.vilnius.tvarkau.views.adapters
+
+import android.content.Context
+import android.content.Intent
+import android.support.v4.view.PagerAdapter
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.bumptech.glide.Glide
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.MarkerOptions
+import kotlinx.android.synthetic.main.problem_images_view_pager_item.view.*
+import kotlinx.android.synthetic.main.problem_images_view_pager_map.view.*
+import lt.vilnius.tvarkau.FullscreenImageActivity
+import lt.vilnius.tvarkau.R
+import lt.vilnius.tvarkau.entity.Problem
+
+
+class ProblemImagesPagerAdapter(context: Context, private val photos: List<String>) : PagerAdapter() {
+
+    private val layoutInflater: LayoutInflater
+    private var problem: Problem?
+
+    constructor(context: Context, problem: Problem) : this(context, problem.photos.orEmpty()) {
+        this.problem = problem
+    }
+
+    init {
+        layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        problem = null
+    }
+
+    override fun getCount(): Int {
+        if (isEmpty()) {
+            return 1
+        }
+
+        return photos.count() + if (problem != null) 1 else 0
+    }
+
+    fun isEmpty(): Boolean = photos.isEmpty() && problem == null
+
+    override fun isViewFromObject(view: View, `object`: Any): Boolean {
+        return view === `object`
+    }
+
+    override fun instantiateItem(container: ViewGroup, position: Int): Any {
+        val itemView: View
+
+        if (isEmpty()) {
+            itemView = instantiateEmptyPhoto(container)
+        } else if (position == count - 1 && problem != null) {
+            itemView = instantiateProblemMap(container)
+        } else {
+            itemView = instantiateProblemPhoto(position, container)
+        }
+
+        container.addView(itemView)
+
+        return itemView
+    }
+
+    fun instantiateProblemMap(container: ViewGroup): View {
+        layoutInflater.inflate(R.layout.problem_images_view_pager_map, container, false).run {
+            if (problem_map_view != null) {
+                problem_map_view.onCreate(null)
+                problem_map_view.getMapAsync {
+                    it.run {
+                        moveCamera(CameraUpdateFactory.newLatLng(problem!!.latLng))
```

Comment:
If someone were to remove if condition in `instantiateItem` this would throw exception. Suggest passing problem directly to this function.